### DISCUSSION TITLE
Get BottomSheet height

### DISCUSSION THREAD

**Discussion by Bartosz-L** [gorhom/react-native-bottom-sheet#1350] [Q&A]
Is there a way to actually get the height of the bottom sheet? I can't find any examples nor solutions. It seems like it's impossible but maybe I'm missing something.

Why am I asking... Beneath my bottom sheet there are few buttons I need to be always visible. The Idea is to dynamically animate their position based on BS height. But no luck so far.

In the docs there is a small animated gif with the cloud icon above the bottom sheet. I wanted to have something similar. Buttons stay on map, but when BS is open they swipe up along with BS.

<img width="239" alt="Zrzut ekranu 2023-04-19 o 15 12 49" src="https://user-images.githubusercontent.com/20401418/233085924-ea422a1c-53ae-418d-8db3-44976b3c7e92.png">

**Comment by thianphat** [comment]
Also looking for an answer to this. Thanks!

**Reply by jaakola-aleksanteri** [reply]
Same

**Reply by balloman** [reply]
See my answer here: https://github.com/gorhom/react-native-bottom-sheet/discussions/1350#discussioncomment-7180774

**Comment by songpola** [comment]
Still looking for ways

**Reply by balloman** [reply]
See my answer here: https://github.com/gorhom/react-native-bottom-sheet/discussions/1350#discussioncomment-7180774

**Comment by Bartosz-L** [comment]
I ended up using [react-native-modalize](https://github.com/jeremybarbet/react-native-modalize) and pass the button as a **HeaderComponent**

**Comment by balloman** [comment]
If anyone is curious on how to do this, I was able to use the animatedPostition prop on the bottom sheet! Just create an animated view and a shared value to store it, then you can connect the position of the bottom sheet to the position of your view.

```ts
const bottomSheetPosition = useSharedValue<number>(0);

<Animated.View style={{ top: bottomSheetPosition }} />

<BottomSheet animatedPosition={bottomSheetPosition} />
```

**Reply by songpola** [reply]
Sorry for the late but thank you very much!
PS. Do import `Animated` from `react-native-reanimated`

**Reply by henrikzabel** [reply]
@balloman the `bottomSheetPosition.value` is always 0 for me. Do you know why this could be the case?

**Reply by balloman** [reply]
To be honest, I'm not sure maybe post an example?

**Reply by henrikzabel** [reply]
I just gave up and used an older implementation with useBottomSheetDynamicSnapPoints. I hope there will be a feature like this directly soon

**Reply by AhmetEnesKCC** [reply]
you can use runOnJs to fix this

```tsx
const heightVal = useSharedValue(0);

  const handleHeight = (val: boolean) => {
    setShowButton(val);
  };

  useDerivedValue(() => {
    runOnJS(handleHeight)(heightVal.value < height * 0.8);
  }, [heightVal]);
  ```

**Reply by spagrhetti** [reply]
So helpful, thank you!

**Reply by szymonbrud** [reply]
PERFECT, thanks!

**Reply by duyrk** [reply]
Thank you, you're a life saver

**Reply by btoo** [reply]
🙇‍♂️🙇‍♂️🙇‍♂️

**Reply by TwickE** [reply]
Thanks @balloman I passed the last 4 hours trying to figure out how to do this🙇‍♂️. I had to do some modifications to your solution like substracting the height of the button to the bottomSheetPosition so it's not behind the BottomSheet.
